envoy/network/transport_socket.h

```diff
@@ -235,9 +235,10 @@ class TransportSocketOptions {
   virtual absl::optional<Network::ProxyProtocolData> proxyProtocolOptions() const PURE;
 
   /**
-   * @return filter state from the downstream request or connection.
+   * @return filter state objects from the downstream request or connection
+   * that are marked as shared with the upstream connection.
    */
-  virtual const StreamInfo::FilterStateSharedPtr& filterState() const PURE;
+  virtual const StreamInfo::FilterState::Objects& filterStateObjects() const PURE;
```

Comment:
I think this function name should be more descriptive, maybe something like `downstreamSharedFilterStateObjects()`.

Reply:
Renamed.